### PR TITLE
Add health check for DNS services

### PR DESCRIPTION
**Pull request by olimpias**
Adds DNS server health check feature

Solves #26

### CODE REVIEW

File: core/condition.go [hunk 1]
```diff
@@ -20,6 +20,11 @@ const (
 	// Values that could replace the placeholder: 127.0.0.1, 10.0.0.1, ...
 	IPPlaceHolder = "[IP]"
 
+	// DNSRCode is a place holder for DNS_RCODE
+	//
+	// Values that could be NOERROR, FORMERR, SERVFAIL, NXDOMAIN, NOTIMP and REFUSED
+	DNSRCode = "[DNS_RCODE]"
```
Comment (TwinProduction) [inline core/condition.go:26]: I know the PR is still market as WIP, but well, here's an early review:

For the sake of consistency, this constant should be appended with `Placeholder`

**Review by TwinProduction — CHANGES_REQUESTED** [review]

**Comment by olimpias** [comment]
Hi @TwinProduction , could you review it?

File: core/condition.go [hunk 2]
```diff
@@ -22,6 +22,11 @@ const (
 	// Values that could replace the placeholder: 127.0.0.1, 10.0.0.1, ...
 	IPPlaceholder = "[IP]"
 
+	// DNSRCodePlaceHolder is a place holder for DNS_RCODE
+	//
+	// Values that could be NOERROR, FORMERR, SERVFAIL, NXDOMAIN, NOTIMP and REFUSED
+	DNSRCodePlaceHolder = "[DNS_RCODE]"
```
Comment (TwinProduction) [inline core/condition.go:28]: `DNSRCodePlaceHolder` should be `DNSRCodePlaceholder`

This one's a bit my fault, I noticed `Placeholder` and `PlaceHolder` were used inconsistently in the code so I just replaced all `PlaceHolder` by `Placeholder`

File: core/types.go [hunk 3]
```diff
@@ -19,6 +19,9 @@ type Result struct {
 	// HTTPStatus is the HTTP response status code
 	HTTPStatus int `json:"status"`
 
+	//
+	DNSRCode string `json:"dnsr_code"`
```
Comment (TwinProduction) [inline core/types.go:23]: Can you add the missing comment and change `json:"dnsr_code"` to `json:"dns-rcode"`?

**Review by TwinProduction — CHANGES_REQUESTED** [review]

File: core/condition.go (hunk 2 above)
Comment (olimpias) [reply core/condition.go]: it's mine, even you mentioned `Placeholder`, I used `PlaceHolder `. 😅

**Review by TwinProduction — APPROVED** [review]

**Comment by TwinProduction** [comment]
Thank you for your contribution @olimpias!

**Comment by TwinProduction** [comment]
Released in [v1.7.0](https://github.com/TwinProduction/gatus/releases/tag/v1.7.0)

**Pull request merged**